ctapipe/io/hdf5merger.py

```diff
@@ -0,0 +1,467 @@
+                    required_nodes.add(table._v_pathname)
+
+        elif node_type is NodeType.ALGORITHM_TEL_GROUP:
+            for kind_group in h5file.root[node]._f_iter_nodes("Group"):
+                for algorithm_group in kind_group._f_iter_nodes("Group"):
+                    required_nodes.add(algorithm_group._v_pathname)
+        else:
+            raise ValueError(f"Unhandled node type: {node_type} of {node}")
+
+    return required_nodes
+
+
+class CannotMerge(IOError):
+    """Raised when trying to merge incompatible files"""
+
+
+def split_h5path(path):
+    """
+    Split a path inside an hdf5 file into parent / child
+    """
+    head, _, tail = path.rstrip("/").rpartition("/")
+    return head, tail
+
+
+class HDF5Merger(Component):
+    """
+    Class to copy / append / merge ctapipe hdf5 files
+    """
+
+    output_path = traits.Path(directory_ok=False).tag(config=True)
+
+    overwrite = traits.Bool(
+        False,
+        help="If true, the ``output_path`` is overwritten in case it exists. See also ``append``",
+    ).tag(config=True)
+
+    append = traits.Bool(
+        False,
+        help="If true, the ``output_path`` is appended to. See also ``overwrite``",
+    ).tag(config=True)
+
+    telescope_events = traits.Bool(
+        True,
+        help="Whether to include telescope-wise data in merged output",
+    ).tag(config=True)
+
+    simulation = traits.Bool(
+        True,
+        help="Whether to include data only known for simulations in merged output",
+    ).tag(config=True)
+
+    true_images = traits.Bool(
+        True,
+        help="Whether to include true images in merged output",
+    ).tag(config=True)
+
+    true_parameters = traits.Bool(
+        True,
+        help="Whether to include parameters calculated on true images in merged output",
+    ).tag(config=True)
+
+    dl1_images = traits.Bool(
+        True,
+        help="Whether to include dl1 images in merged output",
+    ).tag(config=True)
+
+    dl1_parameters = traits.Bool(
+        True,
+        help="Whether to include dl1 image parameters in merged output",
+    ).tag(config=True)
+
+    dl2_subarray = traits.Bool(
+        True, help="Whether to include dl2 subarray-event-wise data in merged output"
+    ).tag(config=True)
+
+    dl2_telescope = traits.Bool(
+        True, help="Whether to include dl2 telescope-event-wise data in merged output"
+    ).tag(config=True)
+
+    monitoring = traits.Bool(
+        True, help="Whether to include monitoring data in merged output"
+    ).tag(config=True)
+
+    processing_statistics = traits.Bool(
+        True, help="Whether to include processing statistics in merged output"
+    ).tag(config=True)
+
+    def __init__(self, output_path=None, **kwargs):
+        # enable using output_path as posarg
+        if output_path not in {None, traits.Undefined}:
+            kwargs["output_path"] = output_path
+
+        super().__init__(**kwargs)
+
+        appending = self.output_path.exists()
```

Comment:
I think it would be more readable, if this was named `output_exists` and
then later 
```python
appending = output_exists and not overwrite
```
or
```python
if overwrite:
    appending = False
else:
    appending = output_exists
```
or just dont use the variable `appending` and do
```python
if output_exists and not overwrite:
# Check files are compatible
```
since its only ever used there.